### PR TITLE
feat: Move Webchat Panel inside the Page section to prevent overlap over Debug Pane

### LINKED ISSUE #6749: Additional Web Chat Integration P1 / P2's
This is a continuation of #6511 to represent the remaining P1 and P2 work items to be done for the Web Chat integration.

### P1's

- [x] Implement a selected state indicator for the log message currently being shown in the inspector pane
- [x] Auto-inspect the latest message in the log pane if there is no message currently inspected

### P2's

- [x] Extend debug panel outside of the authoring canvas -- it should be at the same level as the project tree
- [ ] Make the Web Chat panel float above / respect the debug panel
- [ ] ~~Implement status bar (Needs investigation)~~

### PR DESCRIPTION
**Pull request by srinaath**
## Description

1.  This PR ensures Webchat panel is nested inside the Page to avoid overlap over the Debug Pane

## Task Item

Fixes #6749 

@stevkan @corinagum Thank you for the insight in spotting the directline channel being closed when webchat component is destroyed!

## Screenshots

https://microsoft-my.sharepoint-df.com/:v:/p/srravich/EbUW-0JvnJZBtklbFvr9ldoBhxqW1Yq4ijQFbfd6aAvgSA?e=hXhASh

### CODE REVIEW

File: Composer/packages/client/src/components/WebChat/WebChatComposer.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+};
+
+export const WebChatComposer = React.memo((props: WebChatComposerProps) => {
+  const { activeLocale, chatData, isDisabled, currentConversation } = props;
+
+  if (!currentConversation || !chatData) {
+    return null;
+  }
+
+  const styleSet = createStyleSet({ ...webChatStyleOptions });
```
Comment (hatpick) [inline Composer/packages/client/src/components/WebChat/WebChatComposer.tsx:91]: can this be moved outside the component to get called once?

File: Composer/packages/client/src/components/WebChat/WebChatContainer.tsx [hunk 2]
```diff
@@ -1,117 +1,41 @@
...
-        const middlewareResult = next(...setupArgs);
-        if (middlewareResult) {
-          return middlewareResult(...renderArgs);
-        }
-        return false;
-      }
-      return false;
-  }
-};
+const webChatPanelZIndex = 1;
```
Comment (hatpick) [inline Composer/packages/client/src/components/WebChat/WebChatContainer.tsx:12]: move this to the zIndex array I've defined for this purpose. The idea is that all z-indices are in the same record so they wont compete with each other.
zIndices.ts

File: Composer/packages/client/src/components/WebChat/WebChatHeader.tsx [hunk 3]
```diff
@@ -33,21 +35,25 @@ const customButtonStyles: IButtonStyles = {
...
+  botName: string;
   currentRestartOption: RestartOption;
   onSetRestartOption: (restartOption: RestartOption) => void;
   conversationId: string;
-  onRestartConversation: (conversationId: string, requireNewUserId: boolean) => Promise<any>;
-  onSaveTranscript: (conversationId: string) => Promise<any>;
+  onRestartConversation: (conversationId: string, requireNewUserId: boolean) => void;
+  onSaveTranscript: (conversationId: string) => void;
   openBotInEmulator: () => void;
+  hideWebChat: () => void;
```
Comment (hatpick) [inline Composer/packages/client/src/components/WebChat/WebChatHeader.tsx:45]: onCloseWebChat: () => void

File: Composer/packages/client/src/components/WebChat/WebChatPanel.tsx [hunk 4]
```diff
@@ -225,10 +228,14 @@ export const WebChatPanel: React.FC<WebChatPanelProps> = ({
...
-    <div ref={webChatPanelRef} style={{ height: 'calc(100% - 38px)' }}>
+    <div ref={webChatPanelRef} style={{ height: `calc(100% - ${webChatHeaderHeight})` }}>
       <WebChatHeader
+        botName={botName}
         conversationId={currentConversation}
         currentRestartOption={currentRestartOption}
+        hideWebChat={() => {
+          setWebChatPanelVisibility(false);
+        }}
         openBotInEmulator={() => {
```
Comment (hatpick) [inline Composer/packages/client/src/components/WebChat/WebChatPanel.tsx:239]: while you are refactoring, please change this to ```onOpenBotInEmulator``` in ```WebChatHeaderProps```

File: Composer/packages/client/src/components/__tests__/useDebugPane.test.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (hatpick) [inline Composer/packages/client/src/components/__tests__/useDebugPane.test.ts:1]: remove one

File: Composer/packages/client/src/components/useDebugPane.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  'dialogs',
+  'language-generation',
+  'language-understanding',
+  'knowledge-base',
+  'botProjectsSettings',
+];
+
+export const useDebugPane = () => {
+  const { location: currentLocation } = useLocation();
+  const [showDebugPane, setVisibilityOfPane] = useState(false);
```
Comment (hatpick) [inline Composer/packages/client/src/components/useDebugPane.ts:18]: follow ```[state, setState]``` naming convention

File: Composer/packages/client/src/components/useDebugPane.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+  useEffect(() => {
+    if (currentLocation?.pathname) {
+      let isShow = false;
+      const result = currentLocation.pathname.replace(/^\/|\/$/g, '').split('/');
```
Comment (hatpick) [inline Composer/packages/client/src/components/useDebugPane.ts:23]: nit: add a comment here explaining what this is supposed to do.

File: Composer/packages/client/src/components/useDebugPane.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+      // URL Patterns /bot/1234.123/dialogs or /bot/1234.123/skill/3434.34/dialogs
+      let pageName = result[2];
+
+      if (result[2] === 'skill') {
+        pageName = result[4];
+      }
+
+      if (pageName) {
+        isShow = !!pagesToShow.some((pattern: string) => {
```
Comment (hatpick) [inline Composer/packages/client/src/components/useDebugPane.ts:33]: !! is not required, some returns true/false

File: Composer/packages/client/src/router.tsx [hunk 9]
```diff
@@ -41,79 +43,118 @@ const Publish = React.lazy(() => import('./pages/publish/Publish'));
...
-            {pluginPages.map((page) => (
-              <PluginPageContainer
-                key={`${page.id}/${page.bundleId}`}
-                bundleId={page.bundleId}
-                path={`plugin/${page.id}/${page.bundleId}`}
-                pluginId={page.id}
+    <div css={root}>
+      <div css={pageWrapper}>
+        <div css={contentWrapper}>
+          <WebChatContainer />
```
Comment (hatpick) [inline Composer/packages/client/src/router.tsx:81]: can it be lazy loaded (at least heavy part of it)? it's only visible on demand

File: Composer/packages/client/src/components/WebChat/WebChatComposer.tsx (hunk 1 above)
Comment (srinaath) [reply Composer/packages/client/src/components/WebChat/WebChatComposer.tsx:91]: I did try this before. On every instantatiation of webchat it keeps resetting back to the old style. Hence dropped it here.

File: Composer/packages/client/src/router.tsx (hunk 9 above)
Comment (srinaath) [reply Composer/packages/client/src/router.tsx:81]: so the actual component is lazy loaded actually. It is shown only when the bot is started and a conversation ID is set. Having this here would basically just spin up a new directline server if none is available. No UI elements to delay the loading of the app until bot is started.

**Review by hatpick — APPROVED** [review]

**Pull request merged**